crates/sui-core/src/consensus_validator.rs

```diff
@@ -58,7 +59,7 @@ impl TransactionValidator for SuiTxValidator {
             match tx.kind {
                 ConsensusTransactionKind::UserTransaction(certificate) => {
                     self.metrics.certificate_signatures_verified.inc();
-                    // todo - verify user signature when we pin signature in certificate
+                    certificate.data().verify()?;
```

Comment:
What does it mean to pin signature?

Reply:
Some time ago hash of certificate contained hash of the signature, e.g. signature was "pinned" into certificate, so we did not need to verify user signature in consensus (since 2f+1 committed that signature is correct)

However, now certificate hash does not include hash of user signature, so one can collect valid certificate, replace a user signature and certificate signature(from validators) will still be correct